src/components/share-drawer/utils.ts

```diff
@@ -0,0 +1,69 @@
+import { ShareModalContent } from 'audius-client/src/common/store/ui/share-modal/types'
+
+import {
+  getCollectionRoute,
+  getTrackRoute,
+  getUserRoute
+} from 'app/utils/routes'
+import { getTwitterLink } from 'app/utils/twitter'
+
+import { messages } from './messages'
+
+export const getContentUrl = content => {
+  switch (content.type) {
+    case 'track': {
+      const { track } = content
+      return getTrackRoute(track, true)
+    }
+    case 'profile': {
+      const { profile } = content
+      return getUserRoute(profile)
+    }
+    case 'album': {
+      const { album, artist } = content
+      return getCollectionRoute({ ...album, user: artist }, true)
+    }
+    case 'playlist': {
+      const { playlist, creator } = content
+      return getCollectionRoute({ ...playlist, user: creator }, true)
+    }
+  }
+}
+
+export const getTwitterShareText = content => {
```

Comment:
Same here - just so we know we need a `ShareModalContent`?

Reply:
Yeah absolutely, we should def get eslint complaining